src/sync/sync_session.cpp

```diff
@@ -455,8 +461,8 @@ std::function<void()> SyncSession::create_notifier_invocation(const NotifierPack
     // A notifier is expired if at least as many bytes have been transferred
     // as were originally considered transferrable.
     is_expired = !package.is_streaming && transferred >= package.captured_transferrable;
-    return [=](){
-        package.notifier(transferred, transferrable);
+    return [=, notifier=package.notifier](){
```

Comment:
What happens if `package` is destroyed after this function returns but before the closure being returned is invoked? If the closure body was `package.notifier()`, would we capture `package` by copy, or the reference to `package` by copy?